src/main/kotlin/org/jqassistant/tooling/intellij/plugin/editor/templates/RuleJqaTemplateFileCreation.kt

```diff
@@ -92,26 +73,63 @@ abstract class RuleJqaTemplateFileCreator(
         } ?: Messages.showErrorDialog(project, "File created but could not be opened.", "Error")
     }
 
-    // Prompts the user for input using a dialog box and returns the input.
-    private fun getUserInput(project: Project, message: String): String? =
-        Messages.showInputDialog(project, message, "Input", Messages.getQuestionIcon())?.trim()
+    // Prompts the user for input and validates the file name recursively.
+    private fun getValidFileName(project: Project, message: String?, initialValue: String?): String? {
+        var input = getUserInput(project, message ?: "Enter file name:", initialValue) ?: return null
+
+        if (input.isEmpty()) {
+            Messages.showWarningDialog(project, "File name can't be empty.", "Invalid Input")
+            return null
+        } else if (!isNameValid(input)) {
+            Messages.showWarningDialog(project, "Invalid file name.", "Invalid Input")
+            input = getValidFileName(project, message, input) ?: return null // Recursive call for re-entry.
+        }
+
+        return input
+    }
 
-    // Validates the file name to ensure it contains only allowed characters and no control characters.
+    // Displays an input dialog and returns the user's input.
+    private fun getUserInput(project: Project, message: String, initialValue: String?): String? =
+        Messages
+            .showInputDialog(
+                project,
+                message,
+                "Input",
+                Messages.getQuestionIcon(),
+                // Pre-fills the input field with the previous value.
+                initialValue,
+                null,
+            )?.trim()
+
+    // Validates the file name to ensure it adheres to naming conventions.
     private fun isNameValid(name: String): Boolean {
         if (name.endsWith(".xml")) {
-            return false
+            return false // Disallow file names ending with ".xml".
         } else {
             return PathUtilRt.isValidFileName(name, true)
```

Comment:
```suggestion
            return PathUtilRt.isValidFileName("$name${getFileExtension()}", true)
```

Because you are using `name` for the check instead of `$targetName${getFileExtension()}`, naming a file `..xml` (a dot before .xml) is disallowed, even though technically the PathUtilRt would alllow this filename (you can try this by naming your rule `.xmll` which will make PathUtilRt check if `.xmll` is valid)